### ISSUE TITLE
Captcha is cached after successful submit

### ISSUE THREAD

**Issue by rafu1987** [maikschneider/bw_captcha#11]
Hey there and thanks for your extension!

When I successfully submit a form and go back to that form, the captcha is still the same as before. In fact, when I reload the page, the captcha does not change. This is probably because the renderAction of the ext:form plugin is cached, but do you think it's possible to always calculate a new captcha on visiting the page with the form on it?

**Comment by maikschneider** [comment]
Hi @rafu1987,

you're right - the captcha does not change on page reload because of the cached action. IMHO this is no problem - the captcha was not solved and is still valid - if the user wants to get a new image to solve, the reload button should be pressed. If I make the action uncached, the plugin would brake the page cache of every page it is included on.

But there is a simple solution if you don't like this behaviour: Just enable the reload button via constants and trigger a click at every page load - something like this should do the trick: 
```
document.querySelectorAll('.captcha__reload').forEach(btn => btn.click());
```
Of cause you could still hide the button via CSS :)

**Comment by rafu1987** [comment]
Hi @maikschneider!
That's a clever solution, I didn't come up with - I will try this! :-)

Unfortunately even if the captcha is solved and the form submitted and i.e. I want to make another contact request, it will be the same captcha as before.

Maybe we could use the afterSubmit hook of ext:form to generate a new captcha after a successful submit? At least then, the captcha will not be the same for further contact requests. :-)

https://docs.typo3.org/c/typo3/cms-form/main/en-us/D/FrontendRendering/Index.html#aftersubmit

**Comment by maikschneider** [comment]
Ah, okay. If there is no finisher configured that redirects or displays a success message, this could indeed be an issue. However the code snippet from above would solve this too, since it's a new request.

But it's a good point: I never thought about removing solved captchas. I'll have a look into this, thanks!

**Comment by maikschneider** [comment]
Solved in [v3.0.0](https://github.com/maikschneider/bw_captcha/releases/tag/v3.0.0)
